astropy/stats/sigma_clipping.py

```diff
@@ -349,10 +349,14 @@ def _sigmaclip_fast(
         if masked:
             result = np.ma.array(data, mask=mask, copy=copy)
         else:
-            if copy:
-                result = data.astype(float, copy=True)
+            if data.dtype.kind != "f":
+                # float array type is needed to insert nans into the array
+                result = data.astype(np.float32)  # also makes a copy
```

Comment:
Would float16 work? 👀 

https://numpy.org/doc/stable/reference/arrays.scalars.html#numpy.float16

Reply:
No.  `int32` can store up to 2,147,483,647 and `int` much more.  `float16` can only store up to 65500.0.

```
>>> np.finfo(np.float16).max
np.float16(65500.0)

>>> np.iinfo(np.int32).max
2147483647
```